airbyte-integrations/connectors/source-insightly/source_insightly/source.py

```diff
@@ -0,0 +1,378 @@
+
+class Leads(IncrementalInsightlyStream):
+    primary_key = "LEAD_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Leads/Search"
+
+
+class Milestones(IncrementalInsightlyStream):
+    primary_key = "MILESTONE_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Milestones/Search"
+
+class Notes(IncrementalInsightlyStream):
+    primary_key = "NOTE_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Notes/Search"
+
+class Opportunities(IncrementalInsightlyStream):
+    primary_key = "OPPORTUNITY_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Opportunities/Search"
+
+class OpportunityProducts(IncrementalInsightlyStream):
+    primary_key = "OPPORTUNITY_ITEM_ID"
+
+    def path(self, **kwargs) -> str:
+        return "OpportunityLineItem/Search"
+
+
+class Organisations(IncrementalInsightlyStream):
+    primary_key = "ORGANISATION_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Organisations/Search"
+
+class PricebookEntries(IncrementalInsightlyStream):
+    primary_key = "PRICEBOOK_ENTRY_ID"
+
+    def path(self, **kwargs) -> str:
+        return "PricebookEntry/Search"
+
+class Pricebooks(IncrementalInsightlyStream):
+    primary_key = "PRICEBOOK_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Pricebook/Search"
+
+class Products(IncrementalInsightlyStream):
+    primary_key = "PRODUCT_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Product/Search"
+
+class Projects(IncrementalInsightlyStream):
+    primary_key = "PROJECT_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Projects/Search"
+
+
+class Prospects(IncrementalInsightlyStream):
+    primary_key = "PROSPECT_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Prospect/Search"
+
+
+class QuoteProducts(IncrementalInsightlyStream):
+    primary_key = "QUOTATION_ITEM_ID"
+
+    def path(self, **kwargs) -> str:
+        return "QuotationLineItem/Search"
+
+class Quotes(IncrementalInsightlyStream):
+    primary_key = "QUOTE_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Quotation/Search"
+
+
+class Tasks(IncrementalInsightlyStream):
+    primary_key = "TASK_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Tasks/Search"
+
+class Tickets(IncrementalInsightlyStream):
+    primary_key = "TICKET_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Ticket/Search"
+
+class Users(IncrementalInsightlyStream):
+    primary_key = "USER_ID"
+
+    def path(self, **kwargs) -> str:
+        return "Users/Search"
+
+
+# Source
+class SourceInsightly(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        try:
+            token = config.get("token")
+            response = requests.get(f"{BASE_URL}Instance", auth=(token,''))
+            response.raise_for_status()
+
+            result = response.json()
+            logger.info(result)
+
+            return True, None
+        except Exception as e:
+            return False, e
```

Comment:
Can you check how other sources do the `check_connection` it works how you're doing but there are cleaner ways.

Reply:
@marcosmarxm any specific connectors you want me to look at?